### ISSUE TITLE
Proper FluentAssertions.NodaTime library

### ISSUE THREAD

**Issue by lundmikkel** [fluentassertions/fluentassertions#1422]
Hi. I would like to use FluentAssertions with [NodaTime](https://nodatime.org/), but the current support is limited to very generic things given by the general types used.

In order to get proper/full support for NodaTime, it would require a new library, say `FluentAssertions.NodaTime`. Are there any plans of supporting that? If not, would you be willing add a new library - like you currently have support for Autofac - supported by the community?

**Comment by dennisdoomen** [comment]
Hi Mikkel. 

No, we have no plans to support `NodaTime`, even though I'm using it in my own projects. If the community can and will support it, sure.

**Comment by weelink** [comment]
I'm currently working on it: https://github.com/weelink/FluentAssertions.NodaTime

I'm going through the most useful classes in NodaTime first and are writing the assertions for it, trying to keep it similar to the assertions for DateTime, DateTimeOffset etc.

If people want to help, just let me know.

**Comment by 0xced** [comment]
Have you looked at the https://github.com/jagrem/fluentassertions.nodatime repository? It looks like some work was already done and a [NuGet package](https://www.nuget.org/packages/FluentAssertions.NodaTime/) is already published.

**Comment by weelink** [comment]
I haven't. That repository hasn't been touched in 2 years and contains just a few assertions.
Not sure how to deal with the NuGet package.

**Comment by weelink** [comment]
It is pretty much done. I just have to double check things.
Once that is ready, I would appreciate it if someone can take a look to see if it is acceptable enough. I'll let you know when that moment arrives.

**Comment by lundmikkel** [comment]
@weelink Are you making your own library, or something that can be officially released as a `FluentAssertions.NodaTime` library? I'll gladly review it, if it is the latter. A publicly supported library would be awesome.

**Comment by weelink** [comment]
@lundmikkel I would like to have it released as a general library. A fresh pair of eyes is greatly appreciated.

I hope to be able to finish it some time this week. Worst case it will be next week. I just need to go through it again, see if I missed anything (most likely) and make sure the failure messages are consistent.

**Comment by weelink** [comment]
I can keep trying to fine tune it, but at some point it might be good enough as the first release.

I tried contacting the owner of [the NuGet package](https://www.nuget.org/packages/FluentAssertions.NodaTime/) and asked if I could take over the NuGet package. I hope he will respond soon.

@lundmikkel When you have time, can you check it out? I went through the classes and created the basic assertions for it. If you know about some useful assertions that I might have missed, please let me know.

**Comment by weelink** [comment]
I haven't got any news about the ownership of the NuGet package yet.

@dennisdoomen Are you willing to take a (quick) look when you have the time and optionally criticize it?

**Comment by dennisdoomen** [comment]
> I tried contacting the owner of [the NuGet package](https://www.nuget.org/packages/FluentAssertions.NodaTime/) and asked if I could take over the NuGet package. I hope he will respond soon.

Did you get any response?

**Comment by weelink** [comment]
As of today I am the owner of the NuGet package! I will try to publish the first version sometime this week.

**Comment by weelink** [comment]
The first release has been published and is available on NuGet: https://www.nuget.org/packages/FluentAssertions.NodaTime/

🥳

**Comment by dennisdoomen** [comment]
Maybe you can Tweet about it. I'll retweet it.

**Comment by weelink** [comment]
Maybe a Twitter account can be useful. I will create one just in case it is needed.

**Comment by weelink** [comment]
Apparently I had a twitter already (https://twitter.com/WeelinkIT). Just made the tweet.
